### ISSUE TITLE
[Bug] JEI Crash when looking at anvil deleted my curio spellbook

### ISSUE THREAD

**Issue by Urias0** [iron431/Irons-Spells-n-Spellbooks#349]
### Observed behaviour

So i was just looking at the teleport scroll in JEI (the arcane anvil tab) game suddenly crashed when i saw the max level (5) got an outofbound error and when i logged back it my spellbook with all my spells in it were gone 

### Expected behaviour

Crash happen but losing the spellbook is harsh :') could be either a bug with curios or JEI

### Steps to reproduce

1.Check teleport in JEI
2.Click the level 3 to see use and go into the arcane anvil tab
3 Look at the spell "level up" 4+4 to 5 
It happened only once , couldn't replicate it 

### Server Type

Single Player

### Crashlog

https://gist.github.com/Urias0/c63be13d06b6cacccfbb80d4d577afc1

### Iron's Spells N Spellbooks version

irons_spellbooks-1.20.1-3.0.1

### Forge version

Forge 47.2.0

### Other mods

AI-Improvements-1.20-0.5.2
appleskin-forge-mc1.20.1-2.5.1
architectury-9.2.14-forge
artifacts-forge-9.3.0
AttributeFix-Forge-1.20.1-21.0.4
balm-forge-1.20.1-7.2.2
Bookshelf-Forge-1.20.1-20.1.9
BotanyPots-Forge-1.20.1-13.0.25
caelus-forge-3.1.0+1.20
charmofundying-forge-6.5.0+1.20.1
cloth-config-11.1.118-forge
Clumps-forge-1.20.1-12.0.0.3
Companion-1.20.1-forge-5.0.1
cookingforblockheads-forge-1.20.1-16.0.3
coroutil-forge-1.20.1-1.3.7
cristellib-1.1.5-forge
curios-forge-5.7.2+1.20.1
DynamicTrees-1.20.1-1.3.0-BETA2
embeddium-0.3.9+mc1.20.1
EnchantingInfuser-v8.0.2-1.20.1-Forge
explorify-v1.3.0-mc1.20
FarmersDelight-1.20.1-1.2.4
framework-forge-1.20.1-0.6.26
geckolib-forge-1.20.1-4.4.2
goblintraders-forge-1.20.1-1.9.3
gravestone-1.20.1-1.0.10
ironchest-1.20.1-14.4.4
irons_spellbooks-1.20.1-3.0.1
jei-1.20.1-forge-15.3.0.4
JustEnoughResources-1.20.1-1.4.0.247
mcw-fences-1.1.1-mc1.20.1forge
moonlight-1.20-2.11.3-forge
MouseTweaks-forge-mc1.20-2.25
player-animation-lib-forge-1.0.2-rc1+1.20
PuzzlesLib-v8.1.17-1.20.1-Forge
Quark-4.0-438
repurposed_structures-7.1.13+1.20.1-forge
skinlayers3d-forge-1.6.2-mc1.20.1
sophisticatedbackpacks-1.20.1-3.20.3.1029
sophisticatedcore-1.20.1-0.6.9.569
Structory_1.20.2_v1.3.4
supplementaries-1.20-2.8.6
waystones-forge-1.20-14.1.3
WaystonesTeleportPets-1.20-1.20.1--1.2
YeetusExperimentus-Forge-2.3.1-build.6+mc1.20.1
YungsApi-1.20-Forge-4.0.4
YungsBetterDungeons-1.20-Forge-4.0.3
YungsBetterEndIsland-1.20-Forge-2.0.5
YungsBetterNetherFortresses-1.20-Forge-2.0.5
YungsBetterStrongholds-1.20-Forge-4.0.3
YungsBetterWitchHuts-1.20-Forge-3.0.3
YungsBetterWitchHuts-1.20-Forge-3.0.3
Zeta-1.0-14
zmedievalmusic-1.20.1-2.1
zombieawareness-1.20.1-1.13.1

**Comment by iron431** [comment]
closing for duplicate of #337
